src/GitHub.VisualStudio/Services/UsageTracker.cs

```diff
@@ -145,8 +135,29 @@ public void IncrementLoginCount()
             SaveUsage(usage);
         }
 
+        void Initialize()
+        {
+            // The services needed by the usage tracker are loaded when they are first needed to
+            // improve the startup time of the extension.
+            if (userSettings == null)
+            {
+                client = uiProvider.GetService<IMetricsService>();
+                connectionManager = uiProvider.GetService<IConnectionManager>();
+                userSettings = uiProvider.GetService<IPackageSettings>();
+                vsservices = uiProvider.GetService<IVSServices>();
```

Comment:
Any or all of these things can trigger a MEF call via `GetExportedValue`, so we need to make sure this is done from the main thread, not a background thread. We'll need `await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();` before these calls, `GetService` is not thread-safe.